### PR TITLE
course_member: add course filter

### PR DESCRIPTION
**Pull request by abeeto**
- added a query param ('course'), so that I can pass in a course id into the endpoint to fetch relevant coursemembers 
- to be able to do task 2 of students spec.

### CODE REVIEW

File: app/filters/course_member.py [hunk 1]
```diff
@@ -24,4 +24,8 @@ def filter_queryset(self, request, queryset, view):
 
             return queryset.order_by(ordering) if ordering else queryset
 
+        course_param = request.query_params.get("course", None)
```
Comment (Opeyem1a) [inline app/filters/course_member.py:27]: Django has a correct was to add filters to an endpoint. See https://www.django-rest-framework.org/api-guide/filtering/

Using query params is not necessarily wrong and the docs will discuss it as an option, but consider that essentially every request to this endpoint will need to at least filter by course. There is currently 0 use cases for ever calling it without filtering for a course.

Due to this, the request should probably be more like `.../course/1/students` and not `.../students?course=1` (do not take these url structures literally, this is just an illustrative example)

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

File: app/filters/course_member.py (hunk 1 above)
Comment (abeeto) [reply app/filters/course_member.py]: In the front end, will the user be able to see the course/[course_id] part in the url.window? I remember something about how users shouldn't be able to switch between courses easily so I'm guessing that part won't be included in the url header.

Comment (Opeyem1a) [reply app/filters/course_member.py]: This is something to be handled with validation once auth is implemented. We will know which user made the request on the server side and can grant or deny access based on that

File: app/management/commands/makefakeclass.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            section_names.append(section.name)
+        if created:
+            self.stdout.write(
+                self.style.SUCCESS(
+                    f'Student "{student.username}" added to course "{course_member.course}" section "{", ".join(section_names)}"'
+                )
+            )
+        return student
+
+    @transaction.atomic
```
Comment (Opeyem1a) [inline app/management/commands/makefakeclass.py:81]: Nice

File: app/views/course_member.py [hunk 3]
```diff
@@ -6,9 +6,28 @@
...
+
 class CourseMemberViewSet(viewsets.ModelViewSet):
     queryset = CourseMember.objects.all()
     serializer_class = CourseMemberSerializer
     pagination_class = ExamplePagination
     ordering_fields = "__all__"
     filter_backends = [FilterStudents]
+
+    @action(detail=False, methods=["get"], url_path="course/(?P<course>[^/.]+)")
+    def by_course(self, request, course):
```
Comment (Opeyem1a) [inline app/views/course_member.py:21]: Maybe rename this to something more like `get_students_in_course`

I also fear I've made this task more difficult by asking for the changed url pattern, I think you did well, but can you record me something walking through this one @action endpoint and why you made the choices you made? Just haven't used it for this purpose before so wanna learn and confirm we're going down a good road

**Review by Opeyem1a — APPROVED** [review]
Generally seems fine, pls get me that explainer video before merge though

**Pull request merged**